python/mdvtools/auth/authutils.py

```
@@ -0,0 +1,359 @@
+                    "can_read": up.can_read,
+                    "can_write": up.can_write,
+                    "is_owner": up.is_owner
+                }
+                for up in user_projects
+            }
+            user_project_cache[user.id] = project_permissions
+
+        active_projects = ProjectService.get_active_projects()
+        active_projects_cache[:] = active_projects
+
+        logger.info(f"Cached users and their project permissions in memory.")
```

Ruff (0.8.2)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)